examples/ble-gatt-server/src/main/java/no/nordicsemi/android/ble/ble_gatt_server/GattService.kt

```diff
@@ -0,0 +1,241 @@
+    }
+
+    private fun enableBleServices() {
+        serverManager = ServerManager(this)
+        serverManager!!.open()
+
+        bleAdvertiseCallback = BleAdvertiser.Callback()
+
+        val bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+        bluetoothManager.adapter.bluetoothLeAdvertiser?.startAdvertising(
+                BleAdvertiser.settings(),
+                BleAdvertiser.advertiseData(),
+                bleAdvertiseCallback!!
+        )
+    }
+
+    private fun disableBleServices() {
+        if (bleAdvertiseCallback != null) {
+            val bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+            bluetoothManager.adapter.bluetoothLeAdvertiser?.stopAdvertising(bleAdvertiseCallback!!)
+            bleAdvertiseCallback = null
+        }
+
+        serverManager?.close()
+        serverManager = null
+    }
+
+    override fun onBind(intent: Intent?): IBinder? {
+        return DataPlane()
+    }
+
+    /**
+     * Functionality available to clients
+     */
+    inner class DataPlane : Binder() {
+        /**
+         * Change the value of the GATT characteristic that we're publishing
+         */
+        fun setValue(value: String) {
+            serverManager?.setValue(value)
+        }
+    }
+
+    companion object {
+        private const val TAG = "gatt-service"
+    }
+
+    class ServerManager internal constructor(val context: Context) : BleServerManager(context) {
+
+        private var clientConnections: ClientConnections? = null
+
+        fun setValue(value: String) {
+            clientConnections?.setValue(value)
+        }
+
+        override fun log(priority: Int, message: String) {
+            if (BuildConfig.DEBUG || priority == Log.ERROR) {
+                Log.println(priority, TAG, message)
+            }
+        }
+
+        override fun initializeServer(): List<BluetoothGattService> {
+            clientConnections = ClientConnections()
+            setServerObserver(clientConnections!!)
+
+            // In this example there's only one service, so singleton list is created.
+            return Collections.singletonList(
+                    service(ServiceProfile.SERVICE_UUID,
+                            characteristic(ServiceProfile.CHARACTERISTIC_UUID,
+                                    BluetoothGattCharacteristic.PROPERTY_READ // properties
+                                            or BluetoothGattCharacteristic.PROPERTY_NOTIFY,
+                                    BluetoothGattCharacteristic.PERMISSION_READ_ENCRYPTED_MITM,  // permissions
+                                    description("A characteristic to be read", false) // descriptors
+                            ))
+            )
+        }
+
+        inner class ClientConnections : ServerObserver {
+
+            private val clientConnections = mutableMapOf<String, ClientConnection>()
+
+            private var currentValue: String? = null
```

Comment:
A server must retain the current value for all characteristics so that may be provided when a new device joins. Alternatively, perhaps a client connection can ask the server for the current value. Couldn't see how to make a `BleManagerGattCallback` do that though...

Reply:
This is handled by the lib. When you set up the GATT server, you may specify whether a characteristic or descriptor should have shared value (visible for all connected devices) or unique per connected device. E.g.:
https://github.com/NordicSemiconductor/Android-BLE-Library/blob/1b836ad846ef86fed35fc93807ff8e1056afd57f/ble/src/main/java/no/nordicsemi/android/ble/BleServerManager.java#L390
and 
https://github.com/NordicSemiconductor/Android-BLE-Library/blob/1b836ad846ef86fed35fc93807ff8e1056afd57f/ble/src/main/java/no/nordicsemi/android/ble/BleServerManager.java#L424
Values are managed by the library, and e.g. returned on Read request from the remote side.

Reply:
Given the use of `sharedCharacteristic`, is there a means of setting the value of the characteristic against the `BleServerManager` somehow?